### PR TITLE
ONPREM-448: Adjust LiveEntry::getStreamName() to use a switch statement based on the value of $liveBackend

### PR DESCRIPTION
**Pull request by navarro967**
This pull request adjusts the `LiveEntry::getStreamName()` method to use a switch statement based on the value of the `$liveBackend` variable. This allows for different behavior depending on the value of `$liveBackend`, specifically for the `LIVE_BACKEND_WOWZA` and `LIVE_BACKEND_LIVE_NG` cases. This change improves the flexibility and maintainability of the code.

### CODE REVIEW

**Comment by david-winder-kaltura** [comment]
I think better will be something like this:

public function getStreamName ()
    {
        $defaultValue = '%i';
        
        $conf = $this->getPartner()->getLiveStreamBroadcastUrlConfigurations(kDataCenterMgr::getCurrentDcId());
        if (isset($conf['stream_name_template']))
        {
            $defaultValue = str_replace('@ENTRY_ID@', $this->getId(), $conf['stream_name_template']);
        }
        
        return $this->getFromCustomData( "streamName", null, $defaultValue);
    }
    
    and then in the onprem configuration we can set 
    stream_name_template = @ENTRY_ID@_%i

**Comment by navarro967** [comment]
@david-winder-kaltura, I see your reasoning for this however will we still be able to configure this platform-wide as well? Configuring individual partners would be less than ideal for the intended use cases.

**Comment by amiras89** [comment]
the function getLiveStreamBroadcastUrlConfigurations() will take both 'broadcast' map and 'live_stream_broadcast_url_configurations' config from partner.
any value for the same key in 'live_stream_broadcast_url_configurations' will override the system level broadcast.ini

it allows to set 'stream_name_template' in broadcast.ini for a system wide settings
and, if needed, override it by placing it under 'live_stream_broadcast_url_configurations' of partner.

@navarro967 @david-winder-kaltura

File: alpha/lib/model/LiveEntry.php [hunk 1]
```diff
@@ -14,6 +14,11 @@ abstract class LiveEntry extends entry
 	const SIMULIVE_CAPABILITY = 'simulive_capability';
 	const LOW_LATENCY_TAG = 'lowlatency';
 
+	const LIVE_BACKEND_PARAM = 'live_backend';
+	const BROADCAST_MAP = 'broadcast';
+	const LIVE_BACKEND_WOWZA = 0;
```
Comment (yossipapi) [inline alpha/lib/model/LiveEntry.php:19]: Why not use string constants like "wowze" & "liveNG" this way you won't have to crosscheck config value 0 to the class to see what it is.

**Review by yossipapi — CHANGES_REQUESTED** [review]

**Comment by navarro967** [comment]
Params changed to fit David and Yossi's concerns:

broadcast.ini:
`stream_name_template = {entryId}_%i`

result:
image

**Comment by navarro967** [comment]
@yossipapi @amiras89 @david-winder-kaltura, please check

**Review by david-winder-kaltura — APPROVED** [review]

**Comment by david-winder-kaltura** [comment]
Please make sure to run full live QA cycle on this after merged and deploy to nvq2

**Review by inbal-ben-david — APPROVED** [review]

**Pull request merged**